bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/ThingHandlerPanel.java

```diff
@@ -0,0 +1,196 @@
+        }
+    }
+
+    /*
+     * Gets the pointer into the panel's log messages ring buffer
+     * and sends the command for the retrieval of the last event_message
+     */
+    private void handleSystemStatusMessage(CaddxMessage message) {
+        // Get the bridge handler
+        CaddxBridgeHandler bridgeHandler = getCaddxBridgeHandler();
+        if (bridgeHandler == null) {
+            return;
+        }
+
+        String pointer = message.getPropertyById("panel_communicator_stack_pointer");
+        communicatorStackPointer = pointer;
+
+        // build map of log message channels to event numbers
+        HashMap<String, String> map = new HashMap<String, String>();
+        map.put(pointer, CaddxBindingConstants.PANEL_LOG_MESSAGE_N_0);
+        bridgeHandler.sendCommand(CaddxBindingConstants.PANEL_LOG_EVENT_REQUEST, pointer);
+        panelLogMessagesMap = map;
+    }
+
+    /*
+     * This function handles the panel log messages.
+     * If the received event_number matches our communication stack pointer then this is the last panel message. The
+     * channel gets updated and the required log message requests are generated for the update of the other log message
+     * channels
+     */
+    private void handleLogEventMessage(CaddxMessage message) {
+        // Get the bridge handler
+        CaddxBridgeHandler bridgeHandler = getCaddxBridgeHandler();
+        if (bridgeHandler == null) {
+            return;
+        }
+
+        String eventNumberString = message.getPropertyById("panel_log_event_number");
+        String eventSizeString = message.getPropertyById("panel_log_event_size");
+
+        // build the message
+        LogEventMessage logEventMessage = new LogEventMessage(message);
+
+        logger.trace("Log_event: {}", logEventMessage);
+
+        // get the channel id from the map
+        HashMap<String, String> logMap = panelLogMessagesMap;
+        if (logMap != null && logMap.containsKey(eventNumberString)) {
+            String id = logMap.get(eventNumberString);
+            ChannelUID channelUID = new ChannelUID(getThing().getUID(), id);
+            updateChannel(channelUID, logEventMessage.toString());
+        }
+
+        if (communicatorStackPointer != null && eventNumberString.equals(communicatorStackPointer)) {
+            HashMap<String, String> map = new HashMap<String, String>();
+
+            int eventNumber = Integer.parseInt(eventNumberString);
+            int eventSize = Integer.parseInt(eventSizeString);
+
+            // Retrieve at maximum the 10 last log messages from the panel
+            int messagesToRetrieve = (eventSize < 10) ? eventSize : 10;
+            for (int i = 1; i < messagesToRetrieve; i++) {
+                eventNumber--;
+                if (eventNumber < 0) {
+                    eventNumber = eventSize;
+                }
+
+                map.put(Integer.toString(eventNumber), "panel_log_message_n_" + Integer.toString(i));
```

Comment:
```suggestion
                map.put(Integer.toString(eventNumber), "panel_log_message_n_" + i);
```

Reply:
Since you mentioned you can't see my comments anymore, I'm commenting on this, so you might see it again.

Reply:
Done